### ISSUE TITLE
hitArea not defined

### ISSUE THREAD

**Issue by spelltwister** [pixijs/pixi-typescript#6]
Hello,

In the InteractionManager.prototype.processInteractive function, there is a test against the display object's hitArea.  There is no other reference to it, but it should probably be on the DisplayObject's definition.  I use this for restricting from where events are raised on a sprite.

``` typescript
static SpriteFromPoints(renderer, graphics, points, bounds, isInteractive): PIXI.Sprite {
        var polygon = graphics.drawPolygon(points);
        var zoneSprite = new PIXI.Sprite(polygon.generateTexture(renderer, 1, (<any>PIXI).SCALE_MODES.DEFAULT));
        zoneSprite.position.x = bounds.x;
        zoneSprite.position.y = bounds.y;
        zoneSprite.interactive = isInteractive || false;
        if (zoneSprite.interactive) {
            (<any>zoneSprite).hitArea = new PIXI.Polygon(points);
        }
        return zoneSprite;
    }
```

**Comment by clark-stevenson** [comment]
Hey @spelltwister !  

I see what you are saying thanks!  

Just a quick question, if we forget TypeScript, is this not a bit weird??  I mean how a unrelated class is adding a dynamic property at runtime?  I know it is valid JS, but even so, is it a mistake on the JS which has been overlooked or intended that way?

**Comment by spelltwister** [comment]
I agree that it seems inappropriate from a design perspective to have this randomly in there.  I'm not currently involved in the project (though I may start contributing), so I'm not sure what the thinking was, I just found some reference about it on a forum somewhere and it works ;p
Are you familiar with the owners of the PIXI project?
Thanks! 

```
 On Saturday, June 20, 2015 4:24 PM, Clark Stevenson <notifications@github.com> wrote:
```

 Hey @spelltwister ! I see what you are saying thanks! Just a quick question, if we forget TypeScript, is this not a bit weird?? I mean how a unrelated class is adding a dynamic property at runtime? I know it is valid JS, but even so, is it a mistake on the JS which has been overlooked or intended that way?—

**Comment by englercj** [comment]
Things like this happen because in JS it is a pattern to consider plugins like mixins. When you include a plugin it modifies the prototype of some object to now include the functionality of the plugin.

In this case hitArea is a interaction plugin feature. It is added to DOs when the interaction plugin is loaded. Similarly [`cacheAsBitmap`](https://github.com/GoodBoyDigital/pixi.js/blob/master/src/extras/cacheAsBitmap.js) adds extra features to DOs, and so do [other extras](https://github.com/GoodBoyDigital/pixi.js/tree/master/src/extras).

It may be confusing because this should be a property of [`interactionTarget`](https://github.com/GoodBoyDigital/pixi.js/blob/master/src/interaction/interactiveTarget.js) but is missing, it should be added there.

**Comment by clark-stevenson** [comment]
Thanks guys! Added. I left it as `any` but not sure if its restricted to shapes, or even just Rectangles.

**Comment by englercj** [comment]
@clark-stevenson Technically it is defined at "anything that implements `contains(x: number, y: number): boolean`". But saying "only shapes" is close enough :)

**Comment by spelltwister** [comment]
Learn something new everyday.  It's amazing to see how active this project is :)

**Comment by clark-stevenson** [comment]
Thanks for your help guys, this is added!
